__tests__/delete-routes.test.js

```diff
@@ -0,0 +1,106 @@
+const request = require("supertest");
+const mockingoose = require("mockingoose");
+const jwtSecret = require("../middleware/token-strategy");
+
+const model = require("../models/Profile");
+const jwt = require("jsonwebtoken");
+const app = require("../server");
+
+describe("DELETE /api/profile", () => {
+  const userId = "test-user-id";
+  it("deletes profile when exists", async () => {
+    const _existingProfile = {
+      _id: "65bf74e4bbc75c11e6e83ce0",
+      name: "",
+      dob: "1998-12-31T23:00:00.000Z",
+      zodiacSign: "",
+      gender: "F",
+      reason: [],
+      location: {},
+      bio: "",
+      photos: [],
+    };
+
+    mockingoose(model).toReturn(_existingProfile, "findOneAndRemove");
+
+    const payload = {
+      _id: "123",
+      userId,
+    };
+    const token = jwt.sign(payload, jwtSecret);
+
+    const response = await request(app)
+      .delete(`/api/profiles/${userId}`)
+      .set("Authorization", `Bearer ${token}`);
+    expect(response.statusCode).toBe(200);
+  });
+
+  it("returns 401 if trying to delete someone else's profile", async () => {
+    const anotherId = "someone-else's-id";
+
+    const _existingProfile = {
```

Comment:
it looks like this profile is the same for all use cases in this file -  you can make just one constant on the top level and use it